### PR TITLE
first refactor, applied for first 5 requierment

### PR DESCRIPTION
**Pull request by RanaSalameh20**

### CODE REVIEW

File: Price Calculator/DiscountInformation.cs
Comment (hayatzsala) [inline Price Calculator/DiscountInformation.cs]: I suggest extracting both enums into separate 2 files

File: Price Calculator/DiscountInformation.cs
Comment (hayatzsala) [inline Price Calculator/DiscountInformation.cs]: it's better to extract these properties into new type like UPCDiscount, and have the value and the code of UPC, and here add property of the new type, 
similar to what you did in Expense

File: Price Calculator/DiscountInformation.cs
Comment (hayatzsala) [inline Price Calculator/DiscountInformation.cs]: here same suggestion as UPC discount, extract into type and declare here property with the new type

File: Price Calculator/Product.cs
Comment (hayatzsala) [inline Price Calculator/Product.cs]: no need to make it static

**Review by hayatzsala — COMMENTED** [review]
Really LOOKS GREAT, Well Done @RanaSalameh , I added small comments and please add .gitignore file, after that will merge it to main, please don't merge them before I re-review, and after update I will update the target branch into main, but it's better to see all changes while review.

File: Price Calculator/Product.cs
Comment (RanaSalameh20) [reply Price Calculator/Product.cs]: Since the Tax applied to all Products, not to specific product so I made it static 
(The client uses this property by Class name not Instance name)

Comment (RanaSalameh20) [reply Price Calculator/Product.cs]: First Requirement: 
=> There is a product, defined by: Name (string), UPC (can be a plain number) and a price.
Company is only doing business in US dollars. Money is calculated at two decimals precision (e.g. $12.34).

There is a flat-rate tax (currently 20%) **added to all products prices**. Tax is mandatory and equal for all products.

Customer wants to be able to specify tax percentage.

Write code which displays base price and price with tax for a product.

**Review by hayatzsala — APPROVED** [review]

**Pull request merged**
